### PR TITLE
[FIX] website_sale: show vat number in website address

### PR DESCRIPTION
**Pull request by Mtaylorr**
Issue:
======
We can't save address in website of latam company (peru for example)

Steps to reproduce the issue:
=============================
- Install ecommerce and peru localization
- Change the company of the website to the peru company
- Go to website add anything to cart and checkout
- Edit the address
- Save
- Required fields missing even though there is no field with red showing
  a warning.

Origin of the issue:
====================
In the template of the address of `l10n_pe_website_sale` we show the
`partner_info` using an xpath on `vat`. But in the common template we
show the VAT number on condition so when that condition is false we will
have some mandatory fields missing because we didn't show them in the
address form.

Solution:
=========
backport of https://github.com/odoo/odoo/commit/3a2eb31de39c9da162ef1bc53deb88a4c6b841f6

opw-4148782

### CODE REVIEW

**Review by nle-odoo — APPROVED** [review]
seems to makes sense to me


side note: the diff is hard to read, but adding ?w=1 to the URL we see that we only remove the `<t t-if=(mode...)">`

**Review by Feyensv — COMMENTED** [review]
Can we have a test please? :pray:

**Comment by Mtaylorr** [comment]
> Member

@Feyensv , can you take a look when you have time? I added a test, Thanks.

File: addons/website_sale/views/templates.xml [hunk 1]
```diff
@@ -1944,24 +1944,19 @@
                                         <t t-if="website._display_partner_b2b_fields()">
                                             <div class="w-100"/>
                                             <t t-set='vat_warning' t-value="'vat' in checkout and checkout['vat'] and not can_edit_vat" />
-                                            <t t-if="(mode == ('new', 'billing') and is_public_order
-                                                or mode == ('edit', 'billing') and partner_id == website_sale_order.partner_id.id)
-                                                and (can_edit_vat or 'vat' in checkout and checkout['vat'])"
-                                            >
```
Comment (Feyensv) [inline addons/website_sale/views/templates.xml:1950]: IIUC, you're removing this `t-if` without any other consideration?
Meaning you see the vat & company name always, regardless of the address type (meaningless for delivery addresses), the company configuration, ...

There is probably something that has to be updated in the `t-if` to consider cases where the vat has to be set and isn't, but it shouldn't be removed like that.

File: addons/l10n_pe_website_sale/tests/test_l10n_pe_website_sale.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.env['product.product'].create({
+            'name': 'Test Product',
+            'standard_price': 70.0,
+            'list_price': 70.0,
+            'website_published': True,
+        })
+        website = cls.env['website'].get_current_website()
+        website.company_id.account_fiscal_country_id = website.company_id.country_id = cls.env.ref('base.pe')
```
Comment (Feyensv) [inline addons/l10n_pe_website_sale/tests/test_l10n_pe_website_sale.py:18]: no need for a setup class if you only have one test.
Also, please add some assertions to make sure the setup is the one you're expecting (i.e. no vat set on the admin partner)

**Review by Feyensv — COMMENTED** [review]
I believe we should be able to find a solution more targeted that doesn't always show the vat & company name.

File: addons/website_sale/views/templates.xml (hunk 1 above)
Comment (Mtaylorr) [reply addons/website_sale/views/templates.xml]: Hello @Feyensv, I changed the fix here to add a variable to force show the vat with latam countries which has the mimimal xml change

**Comment by Mtaylorr** [comment]
@Feyensv can you take a look at this again ? thanks

**Comment by Feyensv** [comment]
Hello @Mtaylorr, sorry for the wait, we were facing the same issue in two versions (17 and 17.4) with two different behaviors (a traceback in 17.4, and being unable to save the address in 17), and I focused on the 17.4 to decide for the best approach on the subject (#183705).

Your approach here, which is to always show the VAT for some specific localizations was one of the approaches considered, but the problem is that child billing address are supposed to have the same VAT as their parent contact.
With your approach, you were able to create a billing address with another VAT, but if the VAT on the parent address was modified, it would replace the child one too.  I believe we only want to show (and require) the VAT on the main address, and use the parent VAT for the child addresses.

Please take a look at the fixup commit I pushed to highlight what I believe is the right solution.

**Comment by Mtaylorr** [comment]
@Feyensv , Thanks for the fixup and I squashed the commits

**Comment by Mtaylorr** [comment]
@robodoo  fw=no

**Comment by Feyensv** [comment]
@robodoo up to saas-17.2

**Comment by Mtaylorr** [comment]
@Feyensv I saw you changed the forward port but didn't r+ it , are you still reviewing it or forgot to r+ it ?

**Comment by Feyensv** [comment]
@robodoo r+

**Comment by JrAdhoc** [comment]
Hi @Mtaylorr !
This PR is causing the website to crash in some cases if l10n_ar_website_sale is installed. 
Task submitted in odoo: 4373610
video for more context: https://drive.google.com/file/d/1bKIHtptA9k3XqKuHK3wjku2bxPfMkuiP/view

cc: @Feyensv @vbe-odoo

**Comment by Feyensv** [comment]
Hello @JrAdhoc, thanks for the feedback and the clear video :+1: 
I don't believe that reverting this fix is the best solution, but I'm working on a fix to solve your specific issue in #189312 .  Feel free to take a look and tell me if you notice another issue with it.

**Pull request closed without merging**